coap-chat/README.md

```diff
@@ -0,0 +1,30 @@
+## About
+
+This application provides a simple command line chat using (g)CoAP as transport.
+It allows to send (short) messages to any (IPv6) address reachable by the node.
+All messages are send to and received for a CoAP resource with path `/chat`. 
+
+## Usage
+
+To send messages use the `chat` shell command, it can be invoked as follows:
+
+```
+chat <destination> <nickname> <message>
```

Comment:
Mh... I know this takes a little away from the simplicity at least implementation-wise, but maybe do a `connect` command that takes the `<destination> <nickname>` parameters (and stores them) and just keep `chat <message>`?

Reply:
I would like to keep it as simple as possible, introducing another command would complicate the work flow. Also `connect` might imply or suggest functionality that is not there. Also you typically want to chat with different user (that is here destination addresses) and not only one specific, okay you might simply use `ff02::1` ...

But again, I aimed for simplicity - for a real chat application you might want to have chat rooms, user discovery and lists and more, but that would also blow up the application. 

This whole thing is a side-product of a small school project, where pupils (with no C experience) were introduced to IoT and RIOT - that's why the main goal also was to keep the code simple and with minimal LOCs.

Reply:
Ok, thanks for the clarification. Then I'm for now fine with the PR as is, but maybe you want to introduce DNS, so the pupils don't need to type (rather non-descript) IPv6 addresses? For the application this would only mean maybe 5 more lines (using `sock_dns`), but of course a little bit more configuration in the backend (configuring the nodes' names, setting RAs to distribute the DNS server, etc).

Reply:
Again, I'd rather not - as you already state it requires infrastructure (DNS Server) and setup. 

During the school project we only had a couple of RIOT nodes (at first) and they send messages via link-local addresses and link-local multicast (`ff02::1`) - which might not be very IoT'ish but at least they had fun 😉 

Later we added a Raspberry Pi (with crosscoap) which forwarded all multicast messages to a webserver backend where all message were put into log and could be access via a webpage.

Reply:
(and in the end you still can use IPv6 addresses as a fallback)

Reply:
Ok. Just a suggestion.

Reply:
(gnarf, why does GitHub not live-update a page -.-)